Subject: [patch] uml remove most devfs_mk_symlink calls.patch
From: blaisorblade_spam@yahoo.it
Patch-mainline: 2.6.11-rc1

Remove uses of devfs_mk_symlink().

We didn't do this before to avoid breaking most user setups, but this patch
should be quite harmless.

I've excluded the hottest part, i.e.  the ubd symlink, while removing the
other; I released a end-user tree with this patch and there was a good number
of people using the symlink rather than the preferred name.  That part will be
merged later, I think.

Since now we have evidence of less and less users using devfs, we think that
it will not cause too much problems.

Acked-by: Jeff Dike <jdike@addtoit.com>
Signed-off-by: Paolo 'Blaisorblade' Giarrusso <blaisorblade_spam@yahoo.it>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Acked-by: Gerd Knorr <kraxel@suse.de>
---
 arch/um/drivers/line.c         |    9 +--------
 arch/um/drivers/mmapper_kern.c |    1 -
 2 files changed, 1 insertion(+), 9 deletions(-)

diff -puN arch/um/drivers/line.c~uml-remove-most-devfs_mk_symlink-calls arch/um/drivers/line.c
--- 25/arch/um/drivers/line.c~uml-remove-most-devfs_mk_symlink-calls	2004-11-18 23:29:10.897199048 -0800
+++ 25-akpm/arch/um/drivers/line.c	2004-11-18 23:29:10.902198288 -0800
@@ -415,8 +415,7 @@ struct tty_driver *line_register_devfs(s
 			 struct tty_operations *ops, struct line *lines,
 			 int nlines)
 {
-	int err, i;
-	char *from, *to;
+	int i;
 	struct tty_driver *driver = alloc_tty_driver(nlines);
 
 	if (!driver)
@@ -436,12 +435,6 @@ struct tty_driver *line_register_devfs(s
 	if (tty_register_driver(driver))
 		panic("line_register_devfs : Couldn't register driver\n");
 
-	from = line_driver->symlink_from;
-	to = line_driver->symlink_to;
-	err = devfs_mk_symlink(from, to);
-	if(err) printk("Symlink creation from /dev/%s to /dev/%s "
-		       "returned %d\n", from, to, err);
-
 	for(i = 0; i < nlines; i++){
 		if(!lines[i].valid) 
 			tty_unregister_device(driver, i);
diff -puN arch/um/drivers/mmapper_kern.c~uml-remove-most-devfs_mk_symlink-calls arch/um/drivers/mmapper_kern.c
--- 25/arch/um/drivers/mmapper_kern.c~uml-remove-most-devfs_mk_symlink-calls	2004-11-18 23:29:10.898198896 -0800
+++ 25-akpm/arch/um/drivers/mmapper_kern.c	2004-11-18 23:29:10.903198136 -0800
@@ -128,7 +128,6 @@ static int __init mmapper_init(void)
 	p_buf = __pa(v_buf);
 
 	devfs_mk_cdev(MKDEV(30, 0), S_IFCHR|S_IRUGO|S_IWUGO, "mmapper");
-	devfs_mk_symlink("mmapper0", "mmapper");
 	return(0);
 }
 
_
